### ISSUE TITLE
is it possible to use the train_unet.py script as a regular ldm?

### ISSUE THREAD

**Issue by drimeF0** [teticio/audio-diffusion#40]
i tried to use your code as normal ldm but the model doesn't converge even after 12 epochs (10000+ data examples)
 ![image](https://github.com/teticio/audio-diffusion/assets/67423319/81ac44b2-4216-4eeb-8aad-d8b163781d86)
 I used these settings:
```py

vae = "vae"
weight_decay = 1e-6
train_batch_size = 32
num_epochs = 100
learning_rate = 5e-5
resolution = (256,256)
train_data_dir = "/content/data/images"
num_train_steps=2000
lr_scheduler = "cosine"
lr_warmup_steps=500
```
other code:
```py
model = UNet2DModel(
    sample_size=latent_resolution,
    in_channels=vqvae.config["latent_channels"],
    out_channels=vqvae.config["latent_channels"],
    layers_per_block=2,
    block_out_channels=(128, 128, 256, 256, 512, 512),
    down_block_types=(
                    "DownBlock2D",
                    "DownBlock2D",
                    "DownBlock2D",
                    "DownBlock2D",
                    "AttnDownBlock2D",
                    "DownBlock2D",
    ),
    up_block_types=(
                    "UpBlock2D",
                    "AttnUpBlock2D",
                    "UpBlock2D",
                    "UpBlock2D",
                    "UpBlock2D",
                    "UpBlock2D",
    ),
)
noise_scheduler = DDIMScheduler(num_train_timesteps=num_train_steps)
```

```py
        clean_images = batch
        clean_images = clean_images.to(model.device)
        clean_images = clean_images * 0.18215

        # Sample noise that we'll add to the images
        noise = torch.randn(clean_images.shape).to(clean_images.device)
        bsz = clean_images.shape[0]
        # Sample a random timestep for each image
        timesteps = torch.randint(0,noise_scheduler.config.num_train_timesteps,(bsz, ),device=clean_images.device,).long()

        # Add noise to the clean images according to the noise magnitude at each timestep
        # (this is the forward diffusion process)
        noisy_images = noise_scheduler.add_noise(clean_images,noise,timesteps)
        

        model_output = model(noisy_images, timesteps)["sample"]
        loss = F.mse_loss(model_output, noise)
        loss.backward()

        torch.nn.utils.clip_grad_norm_(model.parameters(), 1.0)
        optimizer.step()
        lr_scheduler.step()
        optimizer.zero_grad()
```
google colab notebook:
https://colab.research.google.com/drive/1AjH6ujmLIsU-4hP0bX0jtaFv_WGHpu2o?usp=sharing

**Comment by drimeF0** [comment]
I noticed that with learning_rate = 1e-4 and lr_scheduler = "cosine" this often leads to NaN loss and black images, even torch.nn.utils.clip_grad_norm_(model.parameters(), 1.0) doesn't help

**Comment by teticio** [comment]
Were you able to get it to work? I didn't find time to dig into your code.
